### ISSUE TITLE
Compute data density by domain and setting (and show in Achilles Web)

### ISSUE THREAD

**Issue by schuemie** [OHDSI/Achilles#118]
We currently have a plot showing data density by domain by calendar time. What would also be very informative is to show data density by domain and by setting (inpatient, outpatient, ER, ...). 

One major difference between DBs is whether they cover inpatient prescribing, or outpatient diagnoses, etc. This plot would help understand these differences, and could be used to decide whether a DB can be used to answer a particular question.

Here's a mockup of what I had in mind:
![densitybysetting](https://cloud.githubusercontent.com/assets/6713328/15647162/37569d50-2662-11e6-8b04-2db81a07e0b8.png)

**Comment by pbr6cornell** [comment]
I think the idea is good in principle, but one thing to point out:  the
foreign key VISIT_OCCURRENCE_ID is optional in all tables, and the only way
to produce such a graph is to assume those optional links are available to
assign each condition/drug/procedure/etc to the VISIT_CONCEPT_ID of its
VISIT table.  In the case of the data that I work with, it's quite likely
that several domains won't be assigned to visits, so they'd all get
classified under 'unknown' even though we may know about the content from
the source meta-data (for example, in US claims data, most drugs are
outpatient pharmacy dispensings, but they aren't assigned to a visit, so
will all be classified as 'unknown' in this rubric).

**Comment by vojtechhuser** [comment]
I agree with this addition as well. In fact I drafted some derived counts in my dev branch. Perhaps just starting simple (with just visit type). 

I like to focus on actual DQ rules/checks and "threshold values" that we would need for such new DQ idea. 

A corresponding rule/DQ check would be that a dataset/DB must have some of visits each visit type. What if the DB/dataset is about ER data only?    
My related issue is formulated here:
http://www.ohdsi.org/web/wiki/doku.php?id=documentation:next_cdm:metadata

**Comment by chrisknoll** [comment]
@vojtechhuser this is an interesting idea.  Let's spin off a new issue related to creating DQ statistics and let this issue focus on the new statistics that are being requested (and the new visualizations that are required on Achilles Web).  I appreciate that your focus is on the DQ elements of Achilles but I also don't want to lose sight that Achilles is also a database characterization tool and so this request can be considered just getting another insight into what is in the data without passing judgement on whether or not it is 'quality' or not.

**Comment by schuemie** [comment]
@pbr6cornell : That would already be something very important to learn from this plot ;-)

**Comment by vojtechhuser** [comment]
@chrisknoll  OK. I will keep that in mind. I will prefix Heel related issues with 'Heel (DQ):'. Derived measures (good for both characterization and DQ) is already a separate issue. The same is true about current analyses - they often have dual purpose.

**Comment by chrisknoll** [comment]
@vojtechhuser , I've created a new label called 'DQ' which you can apply to issues about data quality.  So, instead of prefixing an issue with DQ, you can just assign the label to the issue, and then you will be able to filter the issues list based on this label.

**Comment by vojtechhuser** [comment]
What is our position on optional columns and whether Achilles should "go after them" at all. 
They offer valuable info and I would favor that we can analyze them (as we wish). The argument that if the info in the column is optional (hence not worth our work) - I wonder what the overal team vote for this problem be.
